pareto/examples/multiobjective_optimization/seismicity_vs_cost_MOO.ipynb

```diff
@@ -0,0 +1,531 @@
+    "assert results_min_risk.solver.status == SolverStatus.ok\n",
+    "\n",
+    "with nostdout():\n",
+    "    feasibility_status = is_feasible(model)\n",
+    "\n",
+    "assert feasibility_status"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cfa3a699-dd91-43d6-b5a3-89512adb6552",
+   "metadata": {},
+   "source": [
+    "We investigate the value of the objective function (minimum subsurface risk) and the corresponding cost:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "880b8561-b5f9-4470-acc0-fece09e40a03",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "min_risk_obj = model.v_Z_SubsurfaceRisk.value\n",
+    "min_risk_cost = model.v_Z.value\n",
+    "\n",
+    "print(\"\\nMinimum risk model\")\n",
+    "print(\"------------------\")\n",
+    "print(f\"Minimum risk achievable: {min_risk_obj} {model.v_Z_SubsurfaceRisk.get_units()}\")\n",
+    "print(f\"Cost incurred with minimum risk: {min_risk_cost} {model.v_Z.get_units()}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e4a23a80-c234-4a57-b31c-7191c6084400",
+   "metadata": {},
+   "source": [
+    "We see that when minimizing subsurface risk, it is possible to achieve an objective function value of zero, but it comes at a prohibitively high cost. Note, however, that it is possible to achieve zero total risk for a much lower cost. The reason the cost reported above is so high is that the optimizer arbitrarily chooses to build many pieces of infrastructure which are not needed, since there is no penalty for doing so.\n",
+    "\n",
+    "Note that is not necessarily feasible to achieve zero overall risk for every produced water network model, but it happens to be feasible with this particular model. \n",
+    "\n",
+    "Next, we implement the epsilon constraint method (and in so doing, demonstrate that it is possible to achieve zero risk with a much lower cost than above). We change the objective function back to minimum cost, and then the epsilon constraint method amounts to solving the optimization problem for different fixed values of subsurface risk: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "50afd436-7bdc-4ead-b6b8-6f220dc29af4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Change objective function back to minimum cost\n",
+    "set_objective(model, Objectives.cost)\n",
+    "\n",
+    "# Create lists for results\n",
+    "fixed_risk_vals = list(range(0, 400, 20))  # Subsurface risk values to iterate over\n",
+    "cost_vals_fixed_risk = list()  # List to store corresponding cost values\n",
+    "\n",
+    "for r in fixed_risk_vals:\n",
+    "    print(\"*\" * 60)\n",
+    "    print(f\"Solving minimum cost model constraining subsurface risk to {r}\")\n",
+    "    print(\"*\" * 60)\n",
+    "    model.v_Z_SubsurfaceRisk.fix(r)  # Fix the subsurface risk to the specified value\n",
```

Comment:
Would suggest adding a constraint instead of this, as now model.v_Z_SubsurfaceRisk also includes the binary, the proposed constraint should be 

```
@model.Constraint()
def fix_risk(model):
      return model.v_Z_SubsurfaceRisk +sum(
                    model.subsurface.y_dist[i, j]
                    for i in model.subsurface.SWDSites
                    for j in prox
                )==r
```

Reply:
I changed the subsurface risk objective function so that this suggestion is not necessary.